### PR TITLE
CAKE-5365 | Replace Parsely API calls with curation-cms

### PR DESCRIPTION
**Pull request by vforge**
Consume https://github.com/Wikia/pandora/pull/7745 instead of Parsely API. Since it's time-sensitive ticket, no major cleanup/refactoring has been done.

@Wikia/cake 
@mszabo-wikia 
@unijuglr

### CODE REVIEW

File: extensions/wikia/Recirculation/RecirculationApiController.class.php [hunk 1]
```diff
@@ -36,15 +36,15 @@ public function getPopularPages() {
 	}
 
 	public function getTrendingFandomArticles() {
-		global $wgParselyApiUrl, $wgParselyApiKey, $wgParselyApiSecret, $wgMemc;
```
Comment (mszabo-wikia) [inline extensions/wikia/Recirculation/RecirculationApiController.class.php:39]: Let's remove those variables from config.

File: extensions/wikia/Recirculation/services/CurationCmsArticleService.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+				'error_message' => $e->getMessage(),
+				'status_code' => $e->getCode(),
+			] );
+
+			throw new WikiaException( self::FAILED_TO_CALL_CURATION_CMS, 500, $e );
+		}
+	}
+
+	private static function getThumbnail( array $imageData ): string {
+		if ( array_key_exists( 'url', $imageData ) && array_key_exists( 'url', $imageData ) && array_key_exists( 'url', $imageData ) ) {
```
Comment (mszabo-wikia) [inline extensions/wikia/Recirculation/services/CurationCmsArticleService.php]: `isset( $imageData['key'], $imageData['width'], $imageData['height'] )`

File: extensions/wikia/Recirculation/services/CurationCmsArticleService.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+			if ( VignetteRequest::isVignetteUrl( $imageData['url'] ) ) {
+				return VignetteRequest::fromUrl( $imageData['url'] )
+					->width( $imageData['width'] )
+					->height( $imageData['height'] )
+					->url();
+			}
+		}
+
+		return array_key_exists( 'url', $imageData ) ? $imageData['url'] : false;
```
Comment (mszabo-wikia) [inline extensions/wikia/Recirculation/services/CurationCmsArticleService.php]: `$imageData['url'] ?? false`

File: extensions/wikia/Recirculation/services/CurationCmsArticleService.php [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+	}
+
+	public function getTrendingFandomArticles( int $limit ): array {
+		$response = $this->doApiRequest( $limit );
+		$data = json_decode( $response->getBody(), true );
+		$posts = [];
+
+		if ( isset( $data['feed'] ) ) {
+			foreach ( $data['feed'] as $post ) {
+				$id = isset( $post['id'] ) ? $post['id'] : false;
```
Comment (mszabo-wikia) [inline extensions/wikia/Recirculation/services/CurationCmsArticleService.php]: `$post['id'] ?? false`

File: extensions/wikia/Recirculation/services/CurationCmsArticleService.php [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+				// IW-2588: Hotfix to prevent showcase URLs from polluting the production cache
+				$articlePath = isset( $post['sourceUrl'] ) ? parse_url( $post['sourceUrl'], PHP_URL_PATH ) : false;
+				$headline = isset( $post['headline'] ) ? $post['headline'] : false;
```
Comment (mszabo-wikia) [inline extensions/wikia/Recirculation/services/CurationCmsArticleService.php]: `$post['headline'] ?? false`

File: extensions/wikia/Recirculation/services/CurationCmsArticleService.php [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+				if ( count( $posts ) >= $limit ) {
+					break;
+				}
+			}
+		}
+
+		return array_values( $posts );
+	}
+
+	private function getArticles( array $query ): array {
```
Comment (mszabo-wikia) [inline extensions/wikia/Recirculation/services/CurationCmsArticleService.php]: This method seems to be unused.

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]

File: extensions/wikia/Recirculation/services/CurationCmsArticleService.php (hunk 6 above)
Comment (vforge) [reply extensions/wikia/Recirculation/services/CurationCmsArticleService.php]: https://github.com/Wikia/app/pull/17629/commits/b1606add2b02d26c376baa638f05d0057745e7f7

File: extensions/wikia/Recirculation/services/CurationCmsArticleService.php (hunk 5 above)
Comment (vforge) [reply extensions/wikia/Recirculation/services/CurationCmsArticleService.php]: https://github.com/Wikia/app/pull/17629/commits/b1606add2b02d26c376baa638f05d0057745e7f7

File: extensions/wikia/Recirculation/services/CurationCmsArticleService.php (hunk 4 above)
Comment (vforge) [reply extensions/wikia/Recirculation/services/CurationCmsArticleService.php]: https://github.com/Wikia/app/pull/17629/commits/b1606add2b02d26c376baa638f05d0057745e7f7

File: extensions/wikia/Recirculation/services/CurationCmsArticleService.php (hunk 3 above)
Comment (vforge) [reply extensions/wikia/Recirculation/services/CurationCmsArticleService.php]: https://github.com/Wikia/app/pull/17629/commits/b1606add2b02d26c376baa638f05d0057745e7f7

File: extensions/wikia/Recirculation/services/CurationCmsArticleService.php (hunk 2 above)
Comment (vforge) [reply extensions/wikia/Recirculation/services/CurationCmsArticleService.php]: https://github.com/Wikia/app/pull/17629/commits/b1606add2b02d26c376baa638f05d0057745e7f7

File: extensions/wikia/Recirculation/RecirculationApiController.class.php (hunk 1 above)
Comment (vforge) [reply extensions/wikia/Recirculation/RecirculationApiController.class.php:39]: https://github.com/Wikia/config/pull/2553

**Review by mszabo-wikia — APPROVED** [review]

File: extensions/wikia/Recirculation/services/CurationCmsArticleService.php [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+	 * @param $limit
+	 * @return ResponseInterface
+	 * @throws WikiaException
+	 */
+	private function doApiRequest( int $limit ): ResponseInterface {
+		$client = new Client( [
+			// Base URI is used with relative requests
+			'base_uri' => $this->baseUrl,
+			// You can set any number of default request options.
+			'timeout' => 2.0,
```
Comment (mklucsarits) [inline extensions/wikia/Recirculation/services/CurationCmsArticleService.php]: Why not use `REQUEST_TIMEOUT_SECONDS`? It's unused otherwise.

**Review by mklucsarits — APPROVED** [review]

File: extensions/wikia/Recirculation/services/CurationCmsArticleService.php (hunk 7 above)
Comment (vforge) [reply extensions/wikia/Recirculation/services/CurationCmsArticleService.php]: https://github.com/Wikia/app/pull/17629/commits/77fb67ddbc0b0a81f338056dddca39e8e87db161

**Pull request merged**